presto-hive/src/main/java/io/prestosql/plugin/hive/orc/OrcDeletedRows.java

```diff
@@ -214,12 +216,20 @@ private void loadValidPositions()
         return deletedRows;
     }
 
-    private static Path createPath(String partitionLocation, AcidInfo.DeleteDeltaInfo deleteDeltaInfo, String fileName)
+    private static Path createPath(AcidInfo acidInfo, AcidInfo.DeleteDeltaInfo deleteDeltaInfo, String fileName)
     {
-        Path directory = new Path(partitionLocation, deleteDeltaSubdir(
+        Path directory = new Path(acidInfo.getPartitionLocation(), deleteDeltaSubdir(
                 deleteDeltaInfo.getMinWriteId(),
                 deleteDeltaInfo.getMaxWriteId(),
                 deleteDeltaInfo.getStatementId()));
+
+        if (acidInfo.isDirectInsert()) {
+            verify(CharMatcher.is('_').countIn(fileName) >= 2, "Filename should contain attemptId");
```

Comment:
going further from this place it look that actually we do not need even `directId` flag in `AcidInfo`. Right? We can just check for attemptId in fileName here.

Reply:
bump

Reply:
I'm not really sure if it's safe to assume that if filename has 2 x _ then second substring is an attemptId

Reply:
I agree, but this is how we recognize them. More precisely, we recognize them by BackgroundHiveSplitLoader.BUCKET_WITH_OPTIONAL_ATTEMPT_ID_PATTERN (so: 2 x underscore, literal "bucket" and some digits). It doesn't hurt any more to use this logic here as well.